lib/operations/replace_one.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const Operation = require('./operation').Operation;
+const updateDocuments = require('./common_functions').updateDocuments;
+
+class ReplaceOneOperation extends Operation {
+  constructor(collection, filter, doc, options) {
+    super(options);
+
+    this.collection = collection;
+    this.filter = filter;
+    this.doc = doc;
+  }
+
+  static get aspects() {
+    return new Set([]);
+  }
+
+  execute(callback) {
+    const coll = this.collection;
+    const filter = this.filter;
+    const doc = this.doc;
+    const options = this.options;
+
+    // Set single document update
+    options.multi = false;
+
+    // Execute update
+    updateDocuments(coll, filter, doc, options, (err, r) => {
```

Comment:
quickly looking at this, can this function be hoisted into a `function replaceCallback(err, result, callback)`? Or does it need to be in the closure?

Reply:
It can be hoisted!